document_multiple_records/static/src/js/document.js

```diff
@@ -70,8 +75,8 @@ var _t = instance.web._t,
                 this.on_attachments_loaded([]);
             }
             else {
-                var dom = [ ['attachment_document_ids.res_model', '=', dataset.model], ['attachment_document_ids.res_id', '=', model_id], ['type', 'in', ['binary', 'url']] ];
-                var ds = new instance.web.DataSetSearch(this, 'ir.attachment', dataset.get_context(), dom);
+                var dom = [ ['related_document', '=', true] ];
```

Comment:
If you did `var dom = [ ['related_document', '=', [dataset.model, model_id]] ];` you wouldn't need to lose context.

In `_get_related_model_documents`, you could get these from `args`